src/tosyntax/FStar.ToSyntax.ToSyntax.fs

```diff
@@ -1475,8 +1473,9 @@ let rec desugar_tycon env (d: AST.decl) quals tcs : (env_t * sigelts) =
                 | [] -> k
                 | _ -> mk (Tm_arrow(typars, mk_Total k)) None se.sigrng in
              { se with sigel = Sig_declare_typ(l, [], t, quals) }
-           | _ -> se in
+           | _ -> failwith "Impossible" in
```

Comment:
Hi Clement, looking at this change. I guess you added it because desugar_abstract_tc always returns a Sig_inductive_typ? Just confirming.